### PR TITLE
6762-6765 upstream Nexenta aclinherit/aclmode fixes

### PR DESCRIPTION
**Pull request by yuripv**
Work by @kcrowenex and @trisk.

### CODE REVIEW

**Comment by yuripankov** [comment]
Note that this breaks several zfs-tests test cases, working on fixing them separately.

**Comment by prakashsurya** [comment]
@zettabot go

**Comment by yuripankov** [comment]
I've done a 24h+ ztest run using zloop.sh and not seeing this, is it really related?

**Comment by ahrens** [comment]
@yuripankov The ztest failure is unrelated (the ACL code you're changing isn't used by ztest anyway).  The zfs-test failures look possibly related to your changes (and it sounds like you're already looking into them):

Tests with results other than PASS that are unexpected:
    FAIL acl/nontrivial/zfs_acl_chmod_delete_001_pos (expected PASS)
    FAIL acl/nontrivial/zfs_acl_chmod_inherit_002_neg (expected PASS)
    FAIL acl/nontrivial/zfs_acl_chmod_inherit_004_pos (expected PASS)
    FAIL acl/nontrivial/zfs_acl_chmod_aclmode_001_pos (expected PASS)

I'm not that familiar with the ACL code.  Can you try to find some reviewers on the mailing lists?

**Comment by yuripankov** [comment]
Will do, thanks, Matt.

**Comment by gwr** [comment]
I reviewed these changes when they were being integrated into illumos-nexenta, in case that helps.

**Comment by ahrens** [comment]
@zettabot go

**Comment by ahrens** [comment]
These changes seem fine to me in principle.  I haven't reviewed them in detail.  @jwk404 could you look at the test suite changes?

**Comment by yuripankov** [comment]
Last push is rebase + some comments-only fixes in the zfs-tests.

File: usr/src/test/zfs-tests/tests/functional/acl/nontrivial/zfs_acl_chmod_inherit_004_pos.ksh [hunk 1]
```diff
@@ -25,125 +25,97 @@
...
 
 function cleanup
 {
-	if [[ -d $basedir ]]; then
-		log_must $RM -rf $basedir
-	fi
+	[[ -d $basedir ]] && log_must $RM -rf $basedir
 }
+
 $ZPOOL upgrade -v
```
Comment (jwk404) [inline usr/src/test/zfs-tests/tests/functional/acl/nontrivial/zfs_acl_chmod_inherit_004_pos.ksh]: lines 52-56 should be deleted

**Comment by jwk404** [comment]
LGTM, just the one small nit.
Thanks for diving into this code, and the cleanup too.

**Comment by yuripv** [comment]
I will update the test cases once https://github.com/openzfs/openzfs/pull/78 is integrated - the changes touch same files.

**Comment by ahrens** [comment]
@yuripv #78 has been integrated; you'll need to rebase on master.

**Comment by yuripv** [comment]
Rebased and fixed the nit @jwk404 mentioned.

**Comment by ahrens** [comment]
@zettabot go

**Comment by yuripv** [comment]
The build seems to be stuck here?

**Comment by ahrens** [comment]
@yuripv It's still making progress (which is not visible in the public Jenkins instance)

**Comment by yuripv** [comment]
ztest failure doesn't seem to be related.

BTW, looking at the zfs-tests results, I think we can mark 'grow_replicas_001_pos' as "should PASS" - fixed in 633aa8079184298dfc2ae6966d0d9cc897712b57.

**Comment by ahrens** [comment]
@yuripv great!  FYI, you can open pull requests against https://github.com/openzfs/openzfs-build too.  But I'll take care of this one.

**Comment by yuripv** [comment]
@ahrens Thank you!

**Pull request closed without merging**